### PR TITLE
Feature to upload project uidl

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -50,16 +56,22 @@ app.post("/upload-uidl", async (req, res) => {
   }
 });
 
-app.get("/fetch-uidl/:fileName", async (req, res) => {
-  const { fileName } = req.params;
+app.get("/fetch-uidl/:fileName/:type", async (req, res) => {
```
Comment (Utwo) [inline src/index.ts]: I think is a better idea to switch filename and type in order to have /fetch-uidl/:type/:fileName. So type will be like a namespace. Also we should check if type is neither equal to "component" or "project"

**Review by Utwo — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (JayaKrishnaNamburu) [reply src/index.ts]: I thought the same, but the problem is these API's are being used in our 3rd party projects. So, we need to change in all those places too. But still seems reasonable. We can change and see what all projects needs change.

**Comment by JayaKrishnaNamburu** [comment]
I just have one doubt utwo, can we save them in 2 sub folders like components in one and projects in another. So, that when a component implementation tries to request a project UIDL. We can throw error since they won’t be in the same path.

**Comment by JayaKrishnaNamburu** [comment]
Like in repl, we can explicitly set. But I am thinking more of from external use cases. But it still not so needed anyways :)

**Comment by Utwo** [comment]
> Like in repl, we can explicitly set. But I am thinking more of from external use cases. But it still not so needed anyways :)

Well, we only act as a save storage for external sources. When somebody will save a component or a project in repl, they will know what type of UIDL they saved.

**Comment by JayaKrishnaNamburu** [comment]
Yes, sounds good. But it still helps for us to know and separate componentUIDL's and projectUIDL's in bucket. Except that no actual use-cases i can think of. You can accept the PR and merge it, i can't do since I opened this one, it doesn't allow me to approve 😄

**Pull request merged**